src/main/java/athena/commands/EditCommand.java

```diff
@@ -34,14 +35,14 @@ public EditCommand(int index, String name, String startTime, String duration, St
      * @param ui       Ui
      */
     @Override
-    public void execute(TaskList taskList, Ui ui) {
+    public void execute(TaskList taskList, Ui ui) throws EditException {
         try {
             taskList.editTask(taskIndex, taskName, taskStartTime, taskDuration, taskDeadline,
                     taskRecurrence, taskImportance, taskNotes);
             ui.printTaskEdited(taskIndex, taskName, taskStartTime, taskDuration, taskDeadline,
                     taskRecurrence, taskImportance, taskNotes);
         } catch (IndexOutOfBoundsException e) {
-            ui.printTaskNotFound(taskIndex);
+            throw new EditException();
```

Comment:
I feel that it would be better to have a more specific exception name.

Reply:
This exception isn't finalized yet as I need to wait for the edit command to be fixed so I can test it.